src/witness/yamlWitness.ml

```diff
@@ -159,7 +179,42 @@ struct
       ) nh []
     in
 
-    let yaml_entries = List.rev yaml_entries in (* reverse to make entries in file in the same order as generation messages *)
+    (* TODO: deduplicate *)
+    let entries = LHT.fold (fun (n, c) local acc ->
+        let loc = Node.location n in
+        match n with
+        | Statement _ when not loc.synthetic && WitnessInvariant.is_invariant_node n ->
+          let context: Invariant.context = {
+            scope=Node.find_fundec n;
+            i = -1;
+            lval=None;
+            offset=Cil.NoOffset;
+            deref_invariant=(fun _ _ _ -> Invariant.none) (* TODO: should throw instead? *)
+          }
+          in
+          begin match Spec.C.invariant context c, Spec.D.invariant context local with
+            | Some c_inv, Some inv ->
+              let loc = Node.location n in
+              (* TODO: group by precondition *)
+              let c_inv = InvariantCil.exp_replace_original_name c_inv in (* cannot be split *)
+              let invs = WitnessUtil.InvariantExp.process_exp inv in
```

Comment:
Is there a particular advantage to divide conjunctions into separate expressions?